// @chakra-ui/styled-system/dist/esm/config/position.js
import { createParser, system } from "../core";
import { positiveOrNegative, t } from "../utils";
var config = {
  position: true,
  pos: t.prop("position"),
  zIndex: {
    property: "zIndex",
    scale: "zIndices"
  },
  inset: t.spaceT(["top", "right", "bottom", "left"]),
  insetX: t.spaceT(["left", "right"]),
  insetInline: t.spaceT("insetInline"),
  insetY: t.spaceT(["top", "bottom"]),
  insetBlock: t.spaceT("insetBlock"),
  top: t.spaceT("top"),
  insetBlockStart: t.spaceT("insetBlockStart"),
  bottom: t.spaceT("bottom"),
  insetBlockEnd: t.spaceT("insetBlockEnd"),
  left: t.spaceT("left"),
  insetInlineStart: t.logical({
    scale: "space",
    property: {
      ltr: "left",
      rtl: "right"
    },
    transform: positiveOrNegative
  }),
  right: t.spaceT("right"),
  insetInlineEnd: t.logical({
    scale: "space",
    property: {
      ltr: "right",
      rtl: "left"
    },
    transform: positiveOrNegative
  })
};
Object.assign(config, {
  insetStart: config.insetInlineStart,
  insetEnd: config.insetInlineEnd
});
/**
 * Types for position CSS properties
 */

export var position = system(config);
export var positionParser = createParser(config);
