// @chakra-ui/styled-system/dist/cjs/config/border.js
"use strict";

exports.__esModule = true;
exports.borderParser = exports.border = void 0;

var _core = require("../core");

var _utils = require("../utils");

var config = {
  border: _utils.t.borders("border"),
  borderWidth: _utils.t.borderWidths("borderWidth"),
  borderStyle: _utils.t.borderStyles("borderStyle"),
  borderColor: _utils.t.colors("borderColor"),
  borderRadius: _utils.t.radii("borderRadius"),
  borderTop: _utils.t.borders("borderTop"),
  borderBlockStart: _utils.t.borders("borderBlockStart"),
  borderTopLeftRadius: _utils.t.radii("borderTopLeftRadius"),
  borderStartStartRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: "borderTopLeftRadius",
      rtl: "borderTopRightRadius"
    }
  }),
  borderEndStartRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: "borderBottomLeftRadius",
      rtl: "borderBottomRightRadius"
    }
  }),
  borderTopRightRadius: _utils.t.radii("borderTopRightRadius"),
  borderStartEndRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: "borderTopRightRadius",
      rtl: "borderTopLeftRadius"
    }
  }),
  borderEndEndRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: "borderBottomRightRadius",
      rtl: "borderBottomLeftRadius"
    }
  }),
  borderRight: _utils.t.borders("borderRight"),
  borderInlineEnd: _utils.t.borders("borderInlineEnd"),
  borderBottom: _utils.t.borders("borderBottom"),
  borderBlockEnd: _utils.t.borders("borderBlockEnd"),
  borderBottomLeftRadius: _utils.t.radii("borderBottomLeftRadius"),
  borderBottomRightRadius: _utils.t.radii("borderBottomRightRadius"),
  borderLeft: _utils.t.borders("borderLeft"),
  borderInlineStart: {
    property: "borderInlineStart",
    scale: "borders"
  },
  borderInlineStartRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: ["borderTopLeftRadius", "borderBottomLeftRadius"],
      rtl: ["borderTopRightRadius", "borderBottomRightRadius"]
    }
  }),
  borderInlineEndRadius: _utils.t.logical({
    scale: "radii",
    property: {
      ltr: ["borderTopRightRadius", "borderBottomRightRadius"],
      rtl: ["borderTopLeftRadius", "borderBottomLeftRadius"]
    }
  }),
  borderX: _utils.t.borders(["borderLeft", "borderRight"]),
  borderInline: _utils.t.borders("borderInline"),
  borderY: _utils.t.borders(["borderTop", "borderBottom"]),
  borderBlock: _utils.t.borders("borderBlock"),
  borderTopWidth: _utils.t.borderWidths("borderTopWidth"),
  borderBlockStartWidth: _utils.t.borderWidths("borderBlockStartWidth"),
  borderTopColor: _utils.t.colors("borderTopColor"),
  borderBlockStartColor: _utils.t.colors("borderBlockStartColor"),
  borderTopStyle: _utils.t.borderStyles("borderTopStyle"),
  borderBlockStartStyle: _utils.t.borderStyles("borderBlockStartStyle"),
  borderBottomWidth: _utils.t.borderWidths("borderBottomWidth"),
  borderBlockEndWidth: _utils.t.borderWidths("borderBlockEndWidth"),
  borderBottomColor: _utils.t.colors("borderBottomColor"),
  borderBlockEndColor: _utils.t.colors("borderBlockEndColor"),
  borderBottomStyle: _utils.t.borderStyles("borderBottomStyle"),
  borderBlockEndStyle: _utils.t.borderStyles("borderBlockEndStyle"),
  borderLeftWidth: _utils.t.borderWidths("borderLeftWidth"),
  borderInlineStartWidth: _utils.t.borderWidths("borderInlineStartWidth"),
  borderLeftColor: _utils.t.colors("borderLeftColor"),
  borderInlineStartColor: _utils.t.colors("borderInlineStartColor"),
  borderLeftStyle: _utils.t.borderStyles("borderLeftStyle"),
  borderInlineStartStyle: _utils.t.borderStyles("borderInlineStartStyle"),
  borderRightWidth: _utils.t.borderWidths("borderRightWidth"),
  borderInlineEndWidth: _utils.t.borderWidths("borderInlineEndWidth"),
  borderRightColor: _utils.t.colors("borderRightColor"),
  borderInlineEndColor: _utils.t.colors("borderInlineEndColor"),
  borderRightStyle: _utils.t.borderStyles("borderRightStyle"),
  borderInlineEndStyle: _utils.t.borderStyles("borderInlineEndStyle"),
  borderTopRadius: _utils.t.radii(["borderTopLeftRadius", "borderTopRightRadius"]),
  borderBottomRadius: _utils.t.radii(["borderBottomLeftRadius", "borderBottomRightRadius"]),
  borderLeftRadius: _utils.t.radii(["borderTopLeftRadius", "borderBottomLeftRadius"]),
  borderRightRadius: _utils.t.radii(["borderTopRightRadius", "borderBottomRightRadius"])
};
Object.assign(config, {
  rounded: config.borderRadius,
  roundedTop: config.borderTopRadius,
  roundedTopLeft: config.borderTopLeftRadius,
  roundedTopRight: config.borderTopRightRadius,
  roundedTopStart: config.borderStartStartRadius,
  roundedTopEnd: config.borderStartEndRadius,
  roundedBottom: config.borderBottomRadius,
  roundedBottomLeft: config.borderBottomLeftRadius,
  roundedBottomRight: config.borderBottomRightRadius,
  roundedBottomStart: config.borderEndStartRadius,
  roundedBottomEnd: config.borderEndEndRadius,
  roundedLeft: config.borderLeftRadius,
  roundedRight: config.borderRightRadius,
  roundedStart: config.borderInlineStartRadius,
  roundedEnd: config.borderInlineEndRadius,
  borderStart: config.borderInlineStart,
  borderEnd: config.borderInlineEnd,
  borderTopStartRadius: config.borderStartStartRadius,
  borderTopEndRadius: config.borderStartEndRadius,
  borderBottomStartRadius: config.borderEndStartRadius,
  borderBottomEndRadius: config.borderEndEndRadius,
  borderStartRadius: config.borderInlineStartRadius,
  borderEndRadius: config.borderInlineEndRadius,
  borderStartWidth: config.borderInlineStartWidth,
  borderEndWidth: config.borderInlineEndWidth,
  borderStartColor: config.borderInlineStartColor,
  borderEndColor: config.borderInlineEndColor,
  borderStartStyle: config.borderInlineStartStyle,
  borderEndStyle: config.borderInlineEndStyle
});
/**
 * The prop types for border properties listed above
 */

var border = (0, _core.system)(config);
exports.border = border;
var borderParser = (0, _core.createParser)(config);
exports.borderParser = borderParser;
