// @chakra-ui/styled-system/dist/cjs/config/layout.js
"use strict";

exports.__esModule = true;
exports.layoutParser = exports.layout = void 0;

var _core = require("../core");

var _utils = require("../utils");

var config = {
  width: _utils.t.sizesT("width"),
  inlineSize: _utils.t.sizesT("inlineSize"),
  height: _utils.t.sizes("height"),
  blockSize: _utils.t.sizes("blockSize"),
  boxSize: _utils.t.sizes(["width", "height"]),
  minWidth: _utils.t.sizes("minWidth"),
  minInlineSize: _utils.t.sizes("minInlineSize"),
  minHeight: _utils.t.sizes("minHeight"),
  minBlockSize: _utils.t.sizes("minBlockSize"),
  maxWidth: _utils.t.sizes("maxWidth"),
  maxInlineSize: _utils.t.sizes("maxInlineSize"),
  maxHeight: _utils.t.sizes("maxHeight"),
  maxBlockSize: _utils.t.sizes("maxBlockSize"),
  d: _utils.t.prop("display"),
  overflow: true,
  overflowX: true,
  overflowY: true,
  display: true,
  verticalAlign: true,
  boxSizing: true
};
Object.assign(config, {
  w: config.width,
  h: config.height,
  minW: config.minWidth,
  maxW: config.maxWidth,
  minH: config.minHeight,
  maxH: config.maxHeight
});
/**
 * Types for layout related CSS properties
 */

var layout = (0, _core.system)(config);
exports.layout = layout;
var layoutParser = (0, _core.createParser)(config);
exports.layoutParser = layoutParser;
