// @chakra-ui/styled-system/dist/cjs/config/others.js
"use strict";

exports.__esModule = true;
exports.othersParser = exports.others = void 0;

var _core = require("../core");

var _directionality = require("../utils/directionality");

var floatTransform = function floatTransform(value, _, props) {
  if (props === void 0) {
    props = {};
  }

  var map = {
    left: "right",
    right: "left"
  };
  return (0, _directionality.getIsRtl)(props) ? map[value] : value;
};

var srOnly = {
  border: "0px",
  clip: "rect(0, 0, 0, 0)",
  width: "1px",
  height: "1px",
  margin: "-1px",
  padding: "0px",
  overflow: "hidden",
  whiteSpace: "nowrap",
  position: "absolute"
};
var srFocusable = {
  position: "static",
  width: "auto",
  height: "auto",
  clip: "auto",
  padding: "0",
  margin: "0",
  overflow: "visible",
  whiteSpace: "normal"
};
var config = {
  animation: true,
  appearance: true,
  visibility: true,
  userSelect: true,
  pointerEvents: true,
  cursor: true,
  resize: true,
  objectFit: true,
  objectPosition: true,
  "float": {
    property: "float",
    transform: floatTransform
  },
  willChange: true,
  filter: true,
  clipPath: true,
  srOnly: {
    property: "&",
    transform: function transform(value) {
      if (value === true) return srOnly;
      if (value === "focusable") return srFocusable;
      return {};
    }
  }
};
var others = (0, _core.system)(config);
exports.others = others;
var othersParser = (0, _core.createParser)(config);
exports.othersParser = othersParser;
