// @chakra-ui/styled-system/dist/cjs/core/index.js
"use strict";

exports.__esModule = true;
var _exportNames = {
  createParser: true,
  system: true,
  compose: true,
  Config: true,
  PropConfig: true,
  Parser: true
};
exports.Parser = exports.PropConfig = exports.Config = exports.compose = exports.system = exports.createParser = void 0;

var _createParser = require("./create-parser");

exports.createParser = _createParser.createParser;

var _createStyleFunction = require("./create-style-function");

Object.keys(_createStyleFunction).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _createStyleFunction[key]) return;
  exports[key] = _createStyleFunction[key];
});

var _createSystem = require("./create-system");

exports.system = _createSystem.system;
exports.compose = _createSystem.compose;

var _parseResponsive = require("./parse-responsive");

Object.keys(_parseResponsive).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _parseResponsive[key]) return;
  exports[key] = _parseResponsive[key];
});

var _types = require("./types");

exports.Config = _types.Config;
exports.PropConfig = _types.PropConfig;
exports.Parser = _types.Parser;
